Enable logd blocking in WebContent

pvollan opened the pull request:
#### 4aa42d865f38d565a6f3d4c62fc5c37e0a4d4066
<pre>
Enable logd blocking in WebContent
<a href="https://bugs.webkit.org/show_bug.cgi?id=255222">https://bugs.webkit.org/show_bug.cgi?id=255222</a>
rdar://106827155

Reviewed by Chris Dumez.

Enable logd blocking in WebContent on iOS. In this configuration, the logs will be forwarded to the
Networking process and logged there with the prefix: &apos;WebContent[pid=%d, processIdentifier=%d]&apos;.
I have verified that signposts are working after this change. This patch also contains a change to
improve performance. Most logging takes place on the main thread in WebContent. This patch moves
most of the logging work to a secondary thread with lower priority. Some validation of IPC
parameters has also been removed on the recipient side, in order to improve performance.

* Source/WTF/wtf/spi/cocoa/OSLogSPI.h:
* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp:
(WebKit::NetworkConnectionToWebProcess::logOnBehalfOfWebContent):
(WebKit::validatedASCIIString): Deleted.
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.h:
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.messages.in:
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::registerLogHook):
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::WebProcess::ensureNetworkProcessConnection):
(WebKit::WebProcess::networkProcessConnectionClosed):
(WebKit::WebProcess::networkProcessConnectionID):
(WebKit::WebProcess::setNetworkProcessConnectionID):
* Source/WebKit/WebProcess/WebProcess.h:

Canonical link: <a href="https://commits.webkit.org/262896@main">https://commits.webkit.org/262896@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9f83eabb48cd3a2e6033a00d1e027fdb4379e9ee

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/2933 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/2998 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/3095 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/4338 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/3345 "Built successfully") 
| | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/23/builds/3107 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/3036 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/2570 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/2961 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/3339 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/2622 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/4130 "Built successfully") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/835 "2 new passes 8 flakes 3 failures") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/2604 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/2465 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/2420 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/2597 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/2655 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/3876 "271 api tests failed or timed out") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/2777 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/3002 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/2411 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/2997 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/2637 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/2607 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/728 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/726 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/2626 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/3075 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/2823 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/857 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
```diff
@@ -662,18 +662,36 @@ static void registerLogHook()
+            String logString = String::fromUTF8(messageString);
+            free(messageString);
+
+            callOnMainRunLoop([logChannel = logChannel.isolatedCopy(), logCategory = logCategory.isolatedCopy(), logString = logString.isolatedCopy(), type] {
```
You don't need to go back to the main thread to send IPC. IPC supports sending IPC off the main thread and it is recommended here since we don't want to slow down the main thread unnecessarily.

cdumez commented on Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
```diff
@@ -662,18 +662,36 @@ static void registerLogHook()
-                return;
-            connection->connection().send(Messages::NetworkConnectionToWebProcess::LogOnBehalfOfWebContent(logChannel, logCategory, logString, type, getpid()), 0);
+        Vector<uint8_t> buffer;
+        buffer.resize(msg->buffer_sz);
```
No need to call resize(), you can pass a parameter to the Vector constructor to give it the appropriate size right away. Pretty sure there is even a constructor where you can pass the data you want to memcpy right away.

cdumez commented on Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
```diff
@@ -662,18 +662,36 @@ static void registerLogHook()
+
+        dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0), [logFormat = logFormat.isolatedCopy(), logChannel = logChannel.isolatedCopy(), logCategory = logCategory.isolatedCopy(), type = type, buffer = WTFMove(buffer), privdata = WTFMove(privdata)] {
+            os_log_message_s msg = { 0 };
+            msg.format = logFormat.utf8().data();
```
On the next line, msg.format becomes a stale pointer.

pvollan replied:
Thanks for these great catches, Chris! I will address these in the upcoming patch.

Thanks for reviewing!

cdumez commented on Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
```diff
@@ -662,18 +662,36 @@ static void registerLogHook()
+        privdata.resize(msg->privdata_sz);
+        memcpy(privdata.data(), msg->privdata, msg->privdata_sz);
+
+        dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0), [logFormat = logFormat.isolatedCopy(), logChannel = logChannel.isolatedCopy(), logCategory = logCategory.isolatedCopy(), type = type, buffer = WTFMove(buffer), privdata = WTFMove(privdata)] {
```
As far as I know (please double check), DISPATCH_QUEUE_PRIORITY_BACKGROUND is a *concurrent* dispatch queue. As a result, I don't think the ordering of our logging is guaranteed anymore?

pvollan replied:
Makes sense, will check!

cdumez replied:
https://developer.apple.com/library/archive/documentation/General/Conceptual/ConcurrencyProgrammingGuide/OperationQueues/OperationQueues.html#//apple_ref/doc/uid/TP40008091-CH102-SW5

cdumez commented on Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
```diff
@@ -662,14 +662,30 @@ static void registerLogHook()
+
+        static NeverDestroyed<Ref<WorkQueue>> queue(WorkQueue::create("Log Queue",  WorkQueue::QOS::Background));
+
+        dispatch_async(queue.get()->dispatchQueue(), [logFormat = logFormat.isolatedCopy(), logChannel = logChannel.isolatedCopy(), logCategory = logCategory.isolatedCopy(), type = type, buffer = WTFMove(buffer), privdata = WTFMove(privdata)] {
```
Note for the future, you would want WTFMove(string).isolatedCopy() for all of these. Otherwise, you're going to end up doing a deep copy unnecessarily.

However, given that I suggested using `CString` earlier, you just need to WTFMove() now, no isolatedCopy().

cdumez replied:
Since you decided to use WorkQueue, I recommend you use WorkQueue::dispatch() instead of dispatch_async().

cdumez commented on Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
```diff
@@ -662,14 +662,30 @@ static void registerLogHook()
-        String logString = String::fromUTF8(messageString);
-        free(messageString);
 
+        String logFormat = String::fromUTF8(msg->format);
```
I would use `CString` for all 3 of these, to avoid `fromUTF8()` and `utf8()` calls.
Can just be something like:
```
CString logFormat(msg->format);
```

cdumez commented on Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
```diff
@@ -662,14 +662,30 @@ static void registerLogHook()
+        Vector<uint8_t> buffer(msg->buffer, msg->buffer_sz);
+        Vector<uint8_t> privdata(msg->privdata, msg->privdata_sz);
+
+        static NeverDestroyed<Ref<WorkQueue>> queue(WorkQueue::create("Log Queue",  WorkQueue::QOS::Background));
```
nit: extra space after comma.

cdumez commented on Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
```diff
@@ -662,14 +662,30 @@ static void registerLogHook()
+            String logString = String::fromUTF8(messageString);
+            free(messageString);
+
             auto* connection = WebProcess::singleton().existingNetworkProcessConnection();
```
You shouldn't be calling `WebProcess::singleton().existingNetworkProcessConnection()` off the main thread as this isn't safe. You're going to have to call it on the main thread and capture a RefPtr<IPC::Connection> in the lambda.

cdumez commented on Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
```diff
@@ -662,14 +662,30 @@ static void registerLogHook()
+            msg.privdata_sz = privdata.size();
+
+            char* messageString = os_log_copy_message_string(&msg);
+            String logString = String::fromUTF8(messageString);
```
This seems unnecessarily inefficient. We first allocate a char array, then construct a String from it (which will copy over the characters), then deallocate both strings, only to send IPC.

My recommendation would be to send the logString as a `Span<const char>`. Span doesn't copy over the characters.

Make sure you only free `messageString` after's you've sent the IPC though.

cdumez commented on Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
```diff
@@ -662,14 +662,30 @@ static void registerLogHook()
     os_log_set_hook(OS_LOG_TYPE_DEFAULT, ^(os_log_type_t type, os_log_message_t msg) {
         if (msg->buffer_sz > 1024)
             return;
-        char* messageString = os_log_copy_message_string(msg);
```
Does the hook get called on the main thread? If so, can use ASSERT() that it is the case? Otherwise, it won't be safe to get the NetworkProcess IPC::Connection.

pvollan replied:
Ah, that is a good catch! The hook will be called on the same thread as the logging took place on. I have addressed this issue in the latest patch.

cdumez commented on Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
```diff
@@ -662,18 +662,47 @@ static void registerLogHook()
+
+        NetworkProcessConnection* connection = nullptr;
+        RefPtr<IPC::Connection> ipcConnection;
+        if (isMainThread())
```
Is this ever called on the main thread?

cdumez replied:
Here are my thoughts:
1. If this never gets called on the main thread, then why create our own work queue?
2.It seems unfortunate to ever dispatch to the main thread. We could probably avoid it by having `WebProcess::ensureNetworkProcessConnection()` set the IPC::Connection::UniqueID of the latest connection to the network process is some global variable (with a global Lock) and use IPC::Connection::send(UniqueID) here directly off the main thread.

pvollan replied:
> Is this ever called on the main thread?

Yes, that is the normal case. The hook will be called on the same thread as the logging took place, and most logging is being submitted from the main thread.

pvollan replied:
> Here are my thoughts:
> 
> 1. If this never gets called on the main thread, then why create our own work queue?
>    2.It seems unfortunate to ever dispatch to the main thread. We could probably avoid it by having `WebProcess::ensureNetworkProcessConnection()` set the IPC::Connection::UniqueID of the latest connection to the network process is some global variable (with a global Lock) and use IPC::Connection::send(UniqueID) here directly off the main thread.

That is a great suggestion. I have adopted this approach in the latest patch.

cdumez commented on Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
```diff
@@ -662,18 +662,47 @@ static void registerLogHook()
+                });
+            }
+
+            free(messageString);
```
This isn't safe in the case where you've called callOnMainRunLoop(). You may free the string before the main thread has had a time to IPC it over.

pvollan replied:
Great catch! This has been addressed in the latest patch.

cdumez commented on Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp:
```diff
@@ -1414,35 +1414,16 @@ void NetworkConnectionToWebProcess::installMockContentFilter(WebCore::MockConten
-}
-
-void NetworkConnectionToWebProcess::logOnBehalfOfWebContent(const String& logChannel, const String& logCategory, const String& logString, uint8_t logType, int32_t pid)
+void NetworkConnectionToWebProcess::logOnBehalfOfWebContent(Vector<char>&& logChannel, Vector<char>&& logCategory, IPC::DataReference&& logString, uint8_t logType, int32_t pid)
```
Any reason we're using an IPC::DataReference for the logString but not for the other 2 strings? Looks to me that we don't need to memcpy those characters to vectors just to log them?

pvollan replied:
Yes, we need to copy the log channel and category strings, since they will be freed after returning from the hook. For the log string, we own it's memory, so we can use an IPC::DataReference there.

cdumez replied:
Sorry, I don't understand this comment. This is coming from IPC so we own the memory no matter what. The question is, do we need to memcpy the characters to a Vector before logging or can we log straight from a IPC::DataReference that points to the IPC memory.

Looking at the code, I couldn't see a good reason for using Vectors here.

pvollan replied:
> Sorry, I don't understand this comment. This is coming from IPC so we own the memory no matter what. The question is, do we need to memcpy the characters to a Vector before logging or can we log straight from a IPC::DataReference that points to the IPC memory.
> 
> Looking at the code, I couldn't see a good reason for using Vectors here.

Ah, I see what you mean. I have changed to IPC::DataReference parameters in the latest patch.

Thanks for reviewing!

cdumez commented on Source/WebKit/WebProcess/WebProcess.cpp:
```diff
@@ -2292,6 +2294,19 @@ RemoteMediaEngineConfigurationFactory& WebProcess::mediaEngineConfigurationFacto
 }
 #endif
 
+std::optional<IPC::Connection::UniqueID> WebProcess::networkProcessConnectionID()
```
No real reason to use std::optional here. ObjectIdentifier already has an invalid state and you can boolean check it.

cdumez commented on Source/WebKit/WebProcess/WebProcess.cpp:
```diff
@@ -2292,6 +2294,19 @@ RemoteMediaEngineConfigurationFactory& WebProcess::mediaEngineConfigurationFacto
+    return m_networkProcessConnectionID;
+}
+
+void WebProcess::setNetworkProcessConnectionID(std::optional<IPC::Connection::UniqueID> uniqueID)
```
I don't think this should be using a std::optional either.

pvollan replied:
Resolved in latest patch.

cdumez commented on Source/WebKit/WebProcess/WebProcess.h:
```diff
@@ -648,6 +651,8 @@ class WebProcess : public AuxiliaryProcess
     String m_uiProcessBundleIdentifier;
     RefPtr<NetworkProcessConnection> m_networkProcessConnection;
+    Lock m_lockNetworkProcessConnectionID;
+    std::optional<IPC::Connection::UniqueID> m_networkProcessConnectionID;
```
missing WTF_GUARDED_BY_LOCK.

pvollan replied:
Fixed!

cdumez commented on Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
```diff
@@ -662,18 +662,33 @@ static void registerLogHook()
-                return;
-            connection->connection().send(Messages::NetworkConnectionToWebProcess::LogOnBehalfOfWebContent(logChannel, logCategory, logString, type, getpid()), 0);
+
+        Vector<char> logFormat(msg->format, msg->format ? strlen(msg->format) + 1 : 0);
```
Why aren't those using CString?

cdumez replied:
You didn't respond to this, I think it'd made the code look a bit nicer to use CString.

pvollan replied:
That is a good point. I have changed to Cstring in the latest version.

Thanks for reviewing!

cdumez commented on Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp:
```diff
@@ -1414,35 +1414,18 @@ void NetworkConnectionToWebProcess::installMockContentFilter(WebCore::MockConten
-        validatedString.mutableData()[i] = ' ';
+    OSObjectPtr<os_log_t> osLogChannel;
+    if (logChannel.data() && logCategory.data()) {
+        auto channel = reinterpret_cast<char*>(const_cast<uint8_t*>(logChannel.data()));
```
Not sure why we're const casting. Pretty sure os_log_create() takes `const char*` in parameter, not `char*`.
We likely want just `reinterpret_cast<const char*>(logChannel.data())`

pvollan replied:
Great catch, fixed.

cdumez commented on Source/WebKit/WebProcess/WebProcess.cpp:
```diff
@@ -1278,6 +1279,7 @@ void WebProcess::networkProcessConnectionClosed(NetworkProcessConnection* connec
 #endif
 
     m_networkProcessConnection = nullptr;
+    setNetworkProcessConnectionID(IPC::Connection::UniqueID());
```
Probably can just pass `{ }`

pvollan replied:
Done!

cdumez commented on Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
```diff
@@ -662,18 +662,35 @@ static void registerLogHook()
+
+            char* messageString = os_log_copy_message_string(&msg);
+            IPC::DataReference logString(reinterpret_cast<uint8_t*>(messageString), strlen(messageString) + 1);
+            IPC::DataReference channel(reinterpret_cast<uint8_t*>(const_cast<char *>(logChannel.data())), logChannel.size());
```
Same comment about the const casting here. IPC::DataReference is a `Span<const uint8t>`, not a `Span<uint8_t`.

cdumez commented on Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
```diff
@@ -662,18 +662,35 @@ static void registerLogHook()
+            char* messageString = os_log_copy_message_string(&msg);
+            IPC::DataReference logString(reinterpret_cast<uint8_t*>(messageString), strlen(messageString) + 1);
+            IPC::DataReference channel(reinterpret_cast<uint8_t*>(const_cast<char *>(logChannel.data())), logChannel.size());
+            IPC::DataReference category(reinterpret_cast<uint8_t*>(const_cast<char *>(logCategory.data())), logCategory.size());
```
ditto.

pvollan replied:
Fixed!

cdumez commented on Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
```diff
@@ -662,18 +662,35 @@ static void registerLogHook()
+
+            auto connectionID = WebProcess::singleton().networkProcessConnectionID();
+            if (connectionID)
+                IPC::Connection::send(connectionID, Messages::NetworkConnectionToWebProcess::LogOnBehalfOfWebContent(channel, category, logString, type, getpid()), 0);
```
channel -> logChannel.bytesInludingNullTerminator()
category -> logCategory.bytesInludingNullTerminator()

cdumez commented on Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
```diff
@@ -662,18 +662,35 @@ static void registerLogHook()
+
+            char* messageString = os_log_copy_message_string(&msg);
+            IPC::DataReference logString(reinterpret_cast<uint8_t*>(messageString), strlen(messageString) + 1);
+            IPC::DataReference channel(reinterpret_cast<const uint8_t*>(logChannel.data()), logChannel.data() ? logChannel.length() + 1 : 0);
```
This is no longer needed now that you use CString, see comment below.

cdumez commented on Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
```diff
@@ -662,18 +662,35 @@ static void registerLogHook()
+            char* messageString = os_log_copy_message_string(&msg);
+            IPC::DataReference logString(reinterpret_cast<uint8_t*>(messageString), strlen(messageString) + 1);
+            IPC::DataReference channel(reinterpret_cast<const uint8_t*>(logChannel.data()), logChannel.data() ? logChannel.length() + 1 : 0);
+            IPC::DataReference category(reinterpret_cast<const uint8_t*>(logCategory.data()), logCategory.data() ? logCategory.length() + 1 : 0);
```
ditto.

pvollan replied:
Great point, fixed in latest patch.

cdumez commented on Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
```diff
@@ -662,18 +662,35 @@ static void registerLogHook()
+            msg.privdata = privdata.data();
+            msg.privdata_sz = privdata.size();
+
+            char* messageString = os_log_copy_message_string(&msg);
```
Per the documentation, this can return NULL. I think we should null check the result to protect against crashes.

pvollan replied:
Ah, good point! Fixed in latest version.

Thanks for reviewing!

cdumez reviewed (commented):
Almost perfect. A couple more comments and then I think we'll be in very good shape.

cdumez reviewed (approved):
LGTM.

I like where we ended up. Looks more efficient and safer than earlier iterations.
The code is also fairly straightforward.

pvollan committed:
Enable logd blocking in WebContent
https://bugs.webkit.org/show_bug.cgi?id=255222
rdar://106827155

Reviewed by Chris Dumez.

Enable logd blocking in WebContent on iOS. In this configuration, the logs will be forwarded to the
Networking process and logged there with the prefix: 'WebContent[pid=%d, processIdentifier=%d]'.
I have verified that signposts are working after this change. This patch also contains a change to
improve performance. Most logging takes place on the main thread in WebContent. This patch moves
most of the logging work to a secondary thread with lower priority. Some validation of IPC
parameters has also been removed on the recipient side, in order to improve performance.

* Source/WTF/wtf/spi/cocoa/OSLogSPI.h:
* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp:
(WebKit::NetworkConnectionToWebProcess::logOnBehalfOfWebContent):
(WebKit::validatedASCIIString): Deleted.
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.h:
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.messages.in:
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::registerLogHook):
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::WebProcess::ensureNetworkProcessConnection):
(WebKit::WebProcess::networkProcessConnectionClosed):
(WebKit::WebProcess::networkProcessConnectionID):
(WebKit::WebProcess::setNetworkProcessConnectionID):
* Source/WebKit/WebProcess/WebProcess.h:

Canonical link: https://commits.webkit.org/262896@main
